user_tools/src/spark_rapids_pytools/rapids/rapids_tool.py

```diff
@@ -387,14 +391,22 @@ class RapidsJarTool(RapidsTool):
     """
 
     def _process_jar_arg(self):
+        jar_path = ''
         tools_jar_url = self.wrapper_options.get('toolsJar')
-        if tools_jar_url is None:
-            tools_jar_url = self.ctxt.get_rapids_jar_url()
-        # download the jar
-        jar_path = self.ctxt.platform.storage.download_resource(tools_jar_url,
-                                                                self.ctxt.get_local_work_dir(),
-                                                                fail_ok=False,
-                                                                create_dir=True)
+        try:
+            tools_jar_url = None
```

Comment:
`tools_jar_url` should not be set to `None` here else the custom jar passed by the user would not be used.

Reply:
good catch, I was using this for testing